wreck: add support to set/get scheduler parameters at runtime

tpatki opened the pull request:
Support for [sched/issue337](https://github.com/flux-framework/flux-sched/issues/337). 

Add `flux wreck sched-params set/get` options to support getting/setting runtime parameters in sched. `flux wreck sched-params get`, `flux wreck sched-params get queue-depth`, and `flux-wreck sched-params get queue-depth,delay-sched' should all work.

tpatki commented:
@dongahn: Changed the commit history/msg here as per your suggestion earlier.

SteVwonder commented:
Thanks @tpatki for putting this together.  Generally LGTM! (With the caveat that my Lua isn't that great). The only thing I noticed is the merge commit.  I believe the normal pattern for flux-framework repos is to rebase to master rather than merge with master, but @garlick or @grondo would know better than me.

grondo commented:
Oh, yes, and please rebase onto @master so we don't have the merge commit. Thanks!

garlick commented:
> If you end up having to refactor a bit in sched, I would suggest moving the rpc to the more flexible dictionary for the payload.

Not to go down a rat hole on this, but if you go that route, it may be a good idea for protocol extensibility to push the diciionary into its own object so that other non-dictionary keys can be added later (for example, a flags parameter):
```json
{ 
  "dict":{ 
    "item":42,
    "foo":"bar",
  },
  "flags":0,
}
```
but as @grondo said, what you have makes sense for now.

tpatki commented:
@grondo: rebased, and made the change you requested for printing individual parameters. I'm working on refactoring the `adjust_for_sched_params` in sched.c (as Dong pointed out in the other PR, we need this for the `delay-sched` argument to to propagate correctly). Please test once I've submitted an updated PR for the same in sched (hopefully will finish it sometime today). 

@grondo, @garlick: I will open another issue to refactor the sched end to support the dictionary payload you both suggested.

tpatki commented:
@dongahn, @SteVwonder, @grondo: 
Addressed your earlier comments, and added a minor update that prints "true" instead of 1 for consistency with the sched side. This can now be tested with the PR submitted to sched.

grondo commented:
Actually the fixes will likely be on the sched side since this is just client code.

Merging.

tpatki commented:
@grondo: Thanks! Yes, we need some error checking around queue_depth, I'll open another issue for this.
